### ISSUE TITLE
Type which implements IEqutable<T> should override Equals but already derives from a type which overrides Equals

### ISSUE THREAD

**Issue by thinker227** [dotnet/roslyn-analyzers#6513]
### Analyzer

**Diagnostic ID**: [CA1067](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1067): `Override Equals when implementing IEquatable`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.103](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

When creating a class `A` which inherits from a class `B`, where `A` implements `IEquatable<A>` and `B` overrides `object.Equals`, CA1067 is reported on the type `A` despite `object.Equals` already having been overwritten. This is still the case if the `object.Equals` override in `B` is marked as `sealed`, in which case it is *impossible* to override the method in `A` and get rid of the diagnostic without suppressing it.

### Steps To Reproduce

1. Create a class which overrides `object.Equals`.
2. Create a class which derives from the previously mentioned class and which implements `IEquatable<T>` for its own type.
3. CA1067 is produced on the latter class.

### Expected behavior
CA1067 is not produced because `object.Equals` has already been overwritten in a parent class.

### Actual behavior
CA1067 is produced.

## Additional context

Sample code which demonstrates the issue:
```cs
// 'Type A should override Equals because it implements IEquatable<T>'
class A : B, IEquatable<A>
{
    public bool Equals(A? other) =>
        throw new NotImplementedException();
}

class B
{
    public sealed override bool Equals(object? obj) =>
        throw new NotImplementedException();
}
```

**Comment by Youssef1313** [comment]
The warning looks correct to me. You should override `Equals(object)` of the base class.
